src/App.tsx

```diff
@@ -1,60 +1,143 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
-import classNames from 'classnames';
+import cn from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getAllUsers, getUserPosts } from './utils/api';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [visiblePosts, setVisiblePosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [loaderActive, setLoadedActive] = useState(false);
+  const [noPostsWarning, setNoPostsWarning] = useState(false);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    setError(false);
+    setLoadedActive(true);
+
+    getAllUsers()
+      .then(setUsers)
+      .catch(() => setError(true))
+      .finally(() => {
+        setLoadedActive(false);
+        setTimeout(() => {
+          setError(false);
+        }, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    setError(false);
+    setLoadedActive(true);
+    setNoPostsWarning(false);
+
+    if (!selectedUserId) {
+      setVisiblePosts([]);
+
+      return;
+    }
+
+    getUserPosts(selectedUserId as number)
+      .then((response: Post[]) => {
+        setVisiblePosts(response);
+
+        if (!response.length && selectedUserId) {
```

Comment:
actually, you could use `!visiblePosts.length && selectedUserId` instead of a separate state in one place you use `noPostsWarning`